### PR TITLE
[glsl.in] Implement field lookup

### PR DESCRIPTION
**Pull request by pjoe**
Also includes some minor cleanup in parser

### CODE REVIEW

**Comment by pjoe** [comment]
Guess this conflicts with #201 

@kvark your call which one to do first (and which one then needs rebasing) 😄

File: src/front/glsl/parser.rs [hunk 1]
```diff
@@ -213,20 +213,35 @@ pomelo! {
...
     }
     postfix_expression ::= function_call;
-    postfix_expression ::= postfix_expression Dot FieldSelection {
-        // TODO
-        return Err(ErrorKind::NotImplemented(".field"))
+    postfix_expression ::= postfix_expression(e) Dot Identifier(i) /* FieldSelection in spec */ {
+        let type_inner = extra.resolve_type(e.expression)?;
+        if let TypeInner::Struct{members} = type_inner {
+            let index = members.iter().position(|m| m.name == Some(i.1.clone()));
+            if let Some(index) = index {
```
Comment (kvark) [inline src/front/glsl/parser.rs:224]: you can shorten this by `let index = members.(...).ok_or(ErrorKind::UnknownField(i.0, i.1))?;`

**Review by kvark — APPROVED** [review]
please rebase

File: src/front/glsl/parser.rs (hunk 1 above)
Comment (pjoe) [reply src/front/glsl/parser.rs]: done

**Comment by pjoe** [comment]
Rebase done :)

**Pull request merged**